### DISCUSSION TITLE
Two way binding dynamic component with parent

### DISCUSSION THREAD

**Discussion by procrates** [nuxt/nuxt#17909] [Questions] [answered]
I am really struggling here.... I am trying to create a dynamic input field and pass the value to the parent... Super simple but I am still struggling... 

Input field component
```
<template>
  <div>
    <label :for="name" class="block text-sm font-medium text-gray-700">
      {{ label }}
    </label>
    <div class="flex mt-1 rounded-md shadow-sm">
      <span
        class="inline-flex items-center px-3 text-gray-500 border border-r-0 border-gray-300  rounded-l-md bg-gray-50 sm:text-sm"
      >
        {{ addOn }}
      </span>
      <input
        type="text"
        :name="name"
        :id="name"
        :disabled="isDisabled"
        class="flex-1 block w-full min-w-0 px-3 py-2 border-gray-300 rounded-none  rounded-r-md focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
        :placeholder="placeholder"
        :value="value"
      />
    </div>
  </div>
</template>
<script lang="ts" setup>
defineProps({
  name: {
    type: String,
    required: true,
  },
  label: {
    type: String,
    required: true,
  },
  addOn: {
    type: String,
    required: true,
  },
  isDisabled: {
    type: Boolean,
    required: false,
    default: () => false,
  },
  placeholder: {
    type: String,
    required: false,
    default: () => "",
  },
  value: {
    type: String,
    require: false,
    default: () => "",
  },
});
</script>
```

Parent
```
<MoleculesInputAddOn
      v-model:value="actualKm"
      name="actual-km"
      label="Actual vehicle km"
      add-on="Km"
      :isDisabled="false"
    />
```

I know with nuxt 2 we need to emit an change event. But with nuxt 3 there is no emitter. I was thinking of using global state. But that seems unnecessary, also i dont want to add any static values as it should be dynamic.

Anyone knows?

Thanks!

**Comment by wobsoriano** [comment, marked as answer]
`defineEmits`

https://vuejs.org/guide/components/events.html#usage-with-v-model

**Reply by procrates** [reply]
Lol i was reading this but was so sure that i did not work for Nuxt 3 becouse eventbus wasnt implemented acording to another post... Actually I tried it based on an example: [v-model-on-component codesandbox](https://codesandbox.io/s/v-model-on-component-pex6s?file=/index.html:697-753) and it worked! Thanks for the clarification that $emits work.
